### DISCUSSION TITLE
Call Livewire function from named blade component slot?

### DISCUSSION THREAD

**Discussion by maxwoerner** [livewire/livewire#4874] [1. Help]
Hi there,

my layout consists of different named components like this:
```html
<body>
  <div>
    {{ $header }}
  </div>
  <div>
    {{ $slot }}
  </div>
  <div>
    {{ $footer }}
  </div>
</body>
````

Inside the main slot, I have Livewire component containing a form that looks like this:

```html
<div>
  <x-slot:header>
    Some Text
  </x-slot:header>
  
  <form wire:submit.prevent="submit">
    Some form fields...
    <button type="submit">Submit</button>
  </form>
  
  <x-slot:footer>
    Some Text
  </x-slot:footer>
</div>
```

I would like to place the submit button within the footer slot like this:
```html
<div>
  <x-slot:header>
    Some Text
  </x-slot:header>
  
  <form wire:submit.prevent="submit">
    Some form fields...
  
  <x-slot:footer>
    <button type="submit">Submit</button>
  </x-slot:footer>

  </form>
</div>
```

However, if I do so, I cannot call the submit function of my Livewire component anymore as the button is placed within the named footer slot.

What would be the best way to achieve this? I was thinking of creating a separate component for the submit button that emit an event to the actual component. Is there any better way? I would really much appreciate your support here.

**Comment by denizozturk** [comment]
Doing this with pure javascript would be more complicated. You need to use alpine.js for this method.

Livewire Blade:
```
<button type="submit" x-on:click="$wire.emitTo('your.livewire.component.path', 'functionName')">Submit</button>
```

Livewire Class:
```
protected $listeners = ['functionName' => 'submitForm'];
// or
protected function getListeners()
{
    return [
    'functionName' => 'submitForm',
    ];
}
```
